collections/src/main/scala/strawman/collection/immutable/LazyList.scala

```diff
@@ -220,6 +221,119 @@ sealed abstract class LazyList[+A] extends LinearSeq[A] with LazyListOps[A, Lazy
     if (this.isEmpty) z
     else tail.foldLeft(op(z, head))(op)
   }
+
+
+  protected def tailDefined: Boolean
+  protected def headDefined: Boolean
+
+  /**
+    * @return a string representation of this collection. Undefined elements are
+    *         represented with `"_"`, an undefined tail is represented with `"?"`,
+    *         and cycles are represented with `"..."`
+    *
+    *         Examples:
+    *
+    *           - `"LazyList(_, ?)"`, a non-empty lazy list, whose head has not been
+    *             evaluated ;
+    *           - `"LazyList(_, 1, _, ?)"`, a lazy list with at least three elements,
+    *             the second one has been evaluated ;
+    *           - `"LazyList(1, 2, 3, ...)"`, an infinite lazy list that contains
+    *             a cycle at the fourth element.
+    */
+  override def toString: String = {
```

Comment:
I finally changed my mind and agreed with @SethTisue [suggestion](https://github.com/scala/collection-strawman/issues/376#issuecomment-364211252) of using different characters to represent undefined elements and undefined tail.